Cargo.toml

```diff
@@ -60,6 +60,7 @@ reqwest = { version = "^0.12", default-features = false, optional = true, featur
 serde = { version = "^1", features = ["derive"], optional = true }
 serde_json = { version = "^1", optional = true }
 url = { version = "^2", optional = true }
+itertools = { version = "0.12.1" }
```

Comment:
Try and allow for version updates to dependencies following the semver standards in general unless there is a specific reason for an individual crate version (so for this it should be "^0.12". And for any dependency that gets added I'm going to ask the same question:

Have you tried compiling this in WASM and confirmed it works? We really want to keep our dependencies down as they can rapidly bloat the WASM bundles and frequently there are weird edge cases in dependencies that prevent it from working.

For minor convenience functions I'd prefer not adding dependencies in here in general for those reasons, but I don't know how big of a problem this one is trying to solve yet.

Reply:
I  used `cargo add`  to add the dependency and I guess that is the default behavior. That said looking a little closer it seems the version spec I used and the one you suggested are almost identical. [Based on the examples here ](https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#specifying-dependencies-from-cratesio) I believe this is how they resolve:
```
"0.12.1" := >=0.12.1, <0.13.0
"^0.12" := >=0.12.0, <0.13.0
```

I believe the passing CI indicates it is working on WASM (correct me if I am wrong on that). But yeah in this case I am only using itertools for their `unique()` method. I mainly grabbed it out to avoid writing it by hand while prototyping, if we end up moving forward with this design I'd be happy to write my own version of `unique`

Reply:
Huh I stand corrected, I assumed that was always a hard version lock. I also forgot that we were actually running the WASM tests in CI so yeah that problem is also covered. I think I would prefer leave itertools out for the single use case though.